Automattic/wp-calypso
# Build: Use calypso-build's default preset to transpile dependencies

#### Changes proposed in this Pull Request

Build: Use calypso-build's default preset to transpile dependencies

Prior to this PR, Calypso was using a dedicated babel config, `babel.dependencies.config.js`, to transpile _some_ of our dependencies in `node_modules`, based on this [rule](https://github.com/Automattic/wp-calypso/blob/b1fa53022454d6a9740277b0b05d4d1d20380bb6/webpack.config.js#L104-L138).

I would like to move aforementioned logic that's taking care of transpiling certain `node_modules` into the `calypso-build` package. The reason for this is that Jetpack (which uses `calypso-build` in its toolchain) has currently [a number of Gutenberg blocks that are broken in IE11](https://github.com/Automattic/jetpack/issues/12511) because of a 3rd party dependency that ships untranspiled code. See https://github.com/Automattic/jetpack/issues/12511#issuecomment-497593885 for how I would like to fix this.
 
This PR drops the dedicated babel config and uses the `default` preset from `@automattic/calypso-build` instead, which is a first step towards moving the dependencies transpiling logic into `calypso-build`. This preempts a number of headaches, such as: How would our config mechanism continue to work if we suddenly were dealing with two different babel configs? In particular, if users decided to specify their own Babel config, how would we use that? Would it be used only to transpile our own code, or also 3rd party dependencies in `node_modules/`? Do we add further config options? (See https://github.com/Automattic/jetpack/issues/12511#issuecomment-497504746 which further illustrates this dilemma)

I think it's in the spirit of `calypso-build` to provide choices over options and thus use the package's `default` preset as the babel config for dependencies. It's already quite similar to `babel.dependencies.config.js; I'm going to add a discussion of the differences, and why I think they don't matter.

#### Testing instructions

```
npm run distclean && npm ci
npm start
```

Verify that the app still builds and runs. In particular, check sections that involve any of the transpiled dependencies (Stats screen for `d3`) still work.

#### Follow-up

I'll follow up with a PR to actually move the depedencies transpilation logic to `calypso-build`.
_Edit: #33496_
